internal/metric/prometheus.go

```diff
@@ -118,18 +121,16 @@ func checkReady(ctx context.Context, api promv1.API, t time.Time) (time.Time, er
 			}
 		}
 
-		if target.LastScrape.Before(t) {
+		// Ensure we have done an additional scrape since completion time
+		if target.LastScrape.Before(t.Add(scrapeInterval)) {
```

Comment:
Went through a few iterations of how to best address this. This seemed to be the cleanest approach to get at what we wanted.

Reply:
Wouldn't this need to be `target.LastScrape.Add(scrapeTimeout).Before(t.Add(scrapeInterval))` to ensure we are considering the time at which the scrape completed rather then when it started? Assuming that `scapeTimeout` is a viable stand in for scrape duration (which the API doesn't expose).